manifests/init.pp

```diff
@@ -24,8 +24,20 @@
   $repo_proxy_username                                 = $nodejs::params::repo_proxy_username,
   $repo_url_suffix                                     = $nodejs::params::repo_url_suffix,
   Array $use_flags                                     = $nodejs::params::use_flags,
+  $mac_provider                                        = undef,
```

Comment:
This isn't the right way to do it. For consistency, the best approach is to create a new variable in params.pp `$darwin_package_provider = 'macports'` then do `$darwin_package_provider = $nodejs::params::darwin_package_provider` in init.pp

Reply:
Oooh good point, I was having issues with the provider not overriding when the package statement was in the params as well, but actually that could be moved to init and the default still set in params.